client/driver/docker.go

```diff
@@ -118,6 +118,27 @@ func createContainer(ctx *ExecContext, task *structs.Task, logger *log.Logger) d
 	logger.Printf("[DEBUG] driver.docker: using %d bytes memory for %s", hostConfig.Memory, task.Config["image"])
 	logger.Printf("[DEBUG] driver.docker: using %d cpu shares for %s", hostConfig.CPUShares, task.Config["image"])
 
+	mode, ok := task.Config["network_mode"]
+        if !ok || mode == "" {
+                // docker default
+                logger.Printf("[WARN] driver.docker: no mode specified for networking, defaulting to bridge")
+                mode = "bridge"
+        }
+
+        // Ignore the container mode for now
+        switch mode {
+        case "default", "bridge", "none":
+                logger.Printf("[DEBUG] driver.docker: using %s as network mode", mode)
+        case "host":
+                if len(task.Resources.Networks) != 0 {
```

Comment:
This is incorrect behavior. If tasks bind to ports not given to them by the scheduler then there is the possibility that the scheduler will place a subsequent job with a constraint on a port onto a client that has a task bound to it (this creates a loss of information for the scheduler).

It is okay to bind to the host network namespace but you must only bind to ports specified in the task.Resources.Networks. This goes for all drivers.

So lets remove that warning.